fffw/encoding/filters.py

```diff
@@ -422,6 +422,12 @@ class Upload(VideoFilter):
     extra_hw_frames: int = param(default=64, init=False)
     device: Device = param(skip=True)
 
+    def __post_init__(self) -> None:
+        if isinstance(self.device, dict):
+            # deserialize back after filter cloning
+            self.device = Device(**self.device)
```

Comment:
Why does cloning serialize the device? Maybe you need to fix a cloning process.

Reply:
You may look at `clone` method: `cls(**asdict(instance))` Recursive is only `asdict()` call. This is why constructing `Device` from dict is necessary.

Other possible implementations:
* don't use `asdict` (shallow copy): cloned instance will keep references to same `Device` instance, which is not what clone should do
* use `attrs` or other serialization silver bullets: imo it is overkill, because `Device` is the only thing that requires deep copy.

Reply:
🆗